src/pytest_dbfixtures/port.py

```diff
@@ -54,31 +56,53 @@ def get_port(ports):
 
 def parse_ports(ports):
     """
-    :param str ports: e.g. '3000', '3000-3100', '3000,3002', '?'
+    :param str|list of tuples and/or  ports:
+        e.g. of str '3000', '3000-3100', '3000,3002', '?'
+        e.g. of list [(20000, 30000), {48889, 50121}]
     :returns: ports set reflecting specifified ports list/range.
     :rtype set
     """
 
-    if ports == '?':
+    random_port = ('?', '', False, None)
+    if ports in random_port:
         return None
 
     port_set = set()
 
-    for p in ports.split(','):
-        if '-' not in p:
-            # single, comma-separated port:
-            try:
-                port_set.add(int(p))
-            except ValueError:
-                raise InvalidPortsDefinition(ports)
-        else:
-            # range of ports:
-            try:
-                start, end = p.split('-')
-            except ValueError:
-                raise InvalidPortsDefinition(ports)
-            if end < start:
-                raise InvalidPortsDefinition(ports)
-            port_set.update(range(int(start), int(end) + 1))
+    if type(ports) is str:
+        for p in ports.split(','):
+            if '-' not in p:
+                # single, comma-separated port:
+                try:
+                    port_set.add(int(p))
+                except ValueError:
+                    raise InvalidPortsDefinition(ports)
+            else:
+                # range of ports:
+                try:
+                    start, end = p.split('-')
+                except ValueError:
+                    raise InvalidPortsDefinition(ports)
+                if end < start:
+                    raise InvalidPortsDefinition(ports)
+                port_set.update(range(int(start), int(end) + 1))
+    else:
+        for element in ports:
+            # tuple, mean range of ports
+            if type(element) is tuple:
+                start, end = element
+                if end < start:
+                    raise InvalidPortsDefinition(ports)
+                port_set.update(range(int(start), int(end)))
+            # set, a series of ports
+            elif type(element) is set:
```

Comment:
Someone might pass a `frozenset` (should work like `set`, but it will be ignored) or any other object (that will be silently ignored). Add an `else` clause raising `InvalidPortsDefinition`. Maybe use `isinstance` instead of `type`.